tests/pytests/integration/minion/test_return_retries.py

```diff
@@ -57,14 +58,15 @@ def test_publish_retry(salt_master, salt_minion_retry, salt_cli, salt_run_cli):
 
 @pytest.mark.slow_test
 @pytest.mark.flaky(max_runs=4)
-def test_pillar_timeout(salt_master_factory):
-    cmd = (
-        sys.executable
-        + ' -c "import time; time.sleep(4.8); print(\'{\\"foo\\": \\"bar\\"}\');"'
-    ).strip()
+def test_pillar_timeout(salt_master_factory, tmp_path):
+    cmd = 'print(\'{"foo": "bar"}\');\n'
+
+    with salt.utils.files.fopen(tmp_path / "script.py", "w") as fp:
+        fp.write(cmd)
```

Comment:
As `cmd` is not really a cmd anymore, maybe makes sense to shrink it a bit:
```suggestion
with salt.utils.files.fopen(tmp_path / "script.py", "w") as fp:
        fp.write('print(\'{"foo": "bar"}\');\n')
```

Reply:
IMHO if this is a backport, we shouldn't diverge from upstream.

Reply:
Since I had to adjust/fix the backported commit to use `sys.executable` I just took the ocasion to create upstream PR containing also the suggestions from Victor: https://github.com/saltstack/salt/pull/68331